coursebuilder/modules/assessments/templates/assessment_page.html

```diff
@@ -82,6 +82,9 @@
               </div>
 
             </div>
+            <div class="course-navigation">
+                <a href="course" class="btn right"> {{ gettext('End') }} </a>
```

Comment:
@Tawakalt here.

Reply:
@chegejames Alright. I tried using the same logic but the variables I needed were unavailable for use.

Reply:
@Tawakalt what variables? The course_outline also contains assessments with their corresponding ids.

Reply:
@chegejames `course_outline` is not accessible from the assessments page. So also is `unit.unit_id`.

Reply:
@Tawakalt Most of the assessments are within a unit and therefore have the course_outline and navigation  available to them. Those outside a unit are treated as being independent such as a final assessment. If you choose to implement navigation for assessments outside a unit, then you have to do all buttons i.e prev, next and end. This will mean that you extend the assessments module independently of the CB.

Reply:
@chegejames `All` the assessments have the `course_outline` as an empty array. The length of `course_outline` in all the assessment pages equals 0. Is that the expected result?

Reply:
@Tawakalt No course_outline is undefined as it was not passed in the context.

Reply:
@Tawakalt to add it, you'll need to modify the assessments module.